pkg/service/update/update.go

```diff
@@ -0,0 +1,145 @@
+package updateBinary
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/charmbracelet/glamour"
+	"go.keploy.io/server/utils"
+	"go.uber.org/zap"
+)
+
+// GitHubRelease holds information about the GitHub release.
+type GitHubRelease struct {
+	TagName string `json:"tag_name"`
+	Body    string `json:"body"`
+}
+
+// updater manages the updating process of the Keploy binary.
+type updater struct {
+	logger *zap.Logger
+}
+
+// NewUpdater initializes a new updater instance.
+func NewUpdater(logger *zap.Logger) Updater {
+	return &updater{
+		logger: logger,
+	}
+}
+
+// Updater defines the contract for updating the Keploy binary.
+type Updater interface {
+	UpdateBinary()
+}
+
+var ErrGitHubAPIUnresponsive = errors.New("GitHub API is unresponsive")
+
+// UpdateBinary initiates the update process for the Keploy binary file.
+func (u *updater) UpdateBinary() {
+	currentVersion := utils.Version
+
+	isDockerCmd := len(os.Getenv("IS_DOCKER_CMD")) > 0
+	if isDockerCmd {
+		u.logger.Info("Please Pull the latest Docker image of Keploy")
+		return
+	}
+
+	releaseInfo, err := utils.GetLatestGitHubRelease()
+	latestVersion := releaseInfo.TagName
+	changelog := releaseInfo.Body
+
+	if err != nil {
```

Comment:
this err statement should be before assigning latestVersion and changelog.

Reply:
Refactored